x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/execution/search/extractor/CompositeKeyExtractor.java

```diff
@@ -55,7 +54,7 @@ public CompositeKeyExtractor(String key, Property property, ZoneId zoneId, DataT
     CompositeKeyExtractor(StreamInput in) throws IOException {
         key = in.readString();
         property = in.readEnum(Property.class);
-        if (in.getVersion().onOrAfter(Version.fromId(INTRODUCING_UNSIGNED_LONG.id))) {
+        if (in.getTransportVersion().onOrAfter(INTRODUCING_UNSIGNED_LONG.transportVersion)) {
```

Comment:
how about migrating VersionCompatibilityChecks.INTRODUCING_UNSIGNED_LONG to TransportVersion?
the INTRODUCING_VERSION_FIELD_TYPE could be migrated too?

Reply:
Hmm, there's a more general thing about migrating `VersionCompatibilityChecks` class generally. I'll add it to the list

Reply:
`VersionCompatibilityChecks` is used in `IndexCompatibility` as well - its not just for transport.